subsys/net/lib/sockets/sockets.c

```diff
@@ -615,9 +615,82 @@ static inline ssize_t z_vrfy_zsock_sendmsg(int sock,
 					   const struct msghdr *msg,
 					   int flags)
 {
-	/* TODO: Create a copy of msg_buf and copy the data there */
+	struct msghdr msg_copy;
+	size_t i;
+	int ret;
+
+	memcpy(&msg_copy, msg, sizeof(msg_copy));
```

Comment:
msg is a pointer to user memory, need to use z_user_from_copy()

Reply:
Yep, and we need to validate the socket descriptor too.